app/scripts/controllers/network/network-controller.js

```diff
@@ -246,23 +257,26 @@ export default class NetworkController extends EventEmitter {
     return rpcUrl;
   }
 
-  setRpcTarget(rpcUrl, chainId, ticker = 'ETH', nickname = '', rpcPrefs) {
-    assert.ok(
-      isPrefixedFormattedHexString(chainId),
-      `Invalid chain ID "${chainId}": invalid hex string.`,
-    );
-    assert.ok(
-      isSafeChainId(parseInt(chainId, 16)),
-      `Invalid chain ID "${chainId}": numerical value greater than max safe value.`,
-    );
+  /**
+   * A method for setting a network provider by UUID.
+   *
+   * @param {string} uuid - the universal unique identifier that corresponds to the network configuration we wish to set as selected.
+   * @returns {Promise<string>} The rpcUrl of the network that was just set
+   */
+  setNetworkTarget(uuid) {
+    let configUUID = uuid;
+    if (typeof uuid === 'object') {
+      configUUID = uuid.uuid;
+    }
+    const targetNetwork = this.networkConfigurations.getState()[configUUID];
+
     this._setProviderConfig({
       type: NETWORK_TYPES.RPC,
-      rpcUrl,
-      chainId,
-      ticker,
-      nickname,
-      rpcPrefs,
+      uuid: configUUID,
```

Comment:
Given that this is the provider config object and not the network config object, should this be `networkConfigurationId`?